test/e2e_node/dra_test.go

```diff
@@ -1664,14 +1666,17 @@ func createHealthTestPodAndClaim(ctx context.Context, f *framework.Framework, dr
 		e2epod.DeletePodOrFail(ctx, f.ClientSet, createdPod.Namespace, createdPod.Name)
 	})
 
-	// Update the Pod's status to include the ResourceClaimStatuses, mimicking the scheduler.
-	podToUpdate, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(ctx, createdPod.Name, metav1.GetOptions{})
-	framework.ExpectNoError(err)
-	podToUpdate.Status.ResourceClaimStatuses = []v1.PodResourceClaimStatus{
-		{Name: claimName, ResourceClaimName: &claimName},
-	}
-	_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).UpdateStatus(ctx, podToUpdate, metav1.UpdateOptions{})
-	framework.ExpectNoError(err, "failed to update Pod status with ResourceClaimStatuses")
+	// Patch the Pod's status to include the ResourceClaimStatuses, mimicking the scheduler.
+	patch, err := json.Marshal(v1.Pod{
+		Status: v1.PodStatus{
+			ResourceClaimStatuses: []v1.PodResourceClaimStatus{
+				{Name: claimName, ResourceClaimName: &claimName},
+			},
+		},
+	})
+	framework.ExpectNoError(err, "failed to marshal patch for Pod status")
+	_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Patch(ctx, createdPod.Name, apitypes.StrategicMergePatchType, patch, metav1.PatchOptions{}, "status")
+	framework.ExpectNoError(err, "failed to patch Pod status with ResourceClaimStatuses")
```

Comment:
This looks good, thank you!

BTW, would it be possible to simply include this Status into the pod , so we can create pod with the required status in one step? That would simplify the code even more and eliminate any possibility for the race.